store/tikv/region_cache_test.go

```diff
@@ -322,32 +371,43 @@ func (s *testRegionCacheSuite) TestRegionEpochAheadOfTiKV(c *C) {
 }
 
 func (s *testRegionCacheSuite) TestDropStoreOnSendRequestFail(c *C) {
-	regionCnt := 999
-	cluster := createClusterWithStoresAndRegions(regionCnt)
+	ts := time.Now().Unix()
+	regionCnt, storeCount := 999, 3
+	cluster := createClusterWithStoresAndRegions(regionCnt, storeCount)
 
 	cache := NewRegionCache(mocktikv.NewPDClient(cluster))
+	defer cache.Close()
 	loadRegionsToCache(cache, regionCnt)
-	c.Assert(len(cache.mu.regions), Equals, regionCnt)
+	c.Assert(workableRegions(cache, cache.mu.regions, ts), Equals, regionCnt)
 
 	bo := NewBackoffer(context.Background(), 1)
 	loc, err := cache.LocateKey(bo, []byte{})
 	c.Assert(err, IsNil)
 
-	// Drop the regions on one store, should drop only 1/3 of the regions.
+	// First two drop attempts just switch peer.
+	for i := 0; i < storeCount-1; i++ {
+		rpcCtx, err := cache.GetRPCContext(bo, loc.Region)
+		c.Assert(err, IsNil)
+		cache.OnSendRequestFail(rpcCtx, errors.New("test error"))
+		c.Assert(workableRegions(cache, cache.mu.regions, ts), Equals, regionCnt+1)
+	}
+
+	// Drop the regions on one store, all region will be drop because they are in three stores.
+	// and fail 3 times hit at each stores.
 	rpcCtx, err := cache.GetRPCContext(bo, loc.Region)
 	c.Assert(err, IsNil)
-	cache.DropStoreOnSendRequestFail(rpcCtx, errors.New("test error"))
-	c.Assert(len(cache.mu.regions), Equals, regionCnt*2/3)
+	cache.OnSendRequestFail(rpcCtx, errors.New("test error"))
+	c.Assert(workableRegions(cache, cache.mu.regions, ts), Equals, 0)
 
 	loadRegionsToCache(cache, regionCnt)
-	c.Assert(len(cache.mu.regions), Equals, regionCnt)
+	c.Assert(len(cache.mu.regions), Equals, regionCnt+1)
```

Comment:
why +1?

Reply:
`cache.LocateKey(bo, []byte{})` will +1, it also need +1 in old test case but they use int div so old case can work without +1

Reply:
maybe I need add more test case :D